packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -439,77 +440,106 @@ describe("SharedTree benchmarks", () => {
+			// Number of local commits to rebase over the inbound bunch
+			const localBranchSizes = isInPerformanceTestingMode ? [10, 100] : [2];
+			// The time taken by each scenario can be broken down into 4 time costs:
+			// 1. Constant factor overhead (we ignore this).
+			// 2. The time taken to rebase inbound commits onto the tip of the trunk.
+			// 3. The time taken to compose all inbound commits from a bunch into a single commit.
+			// 4. The time taken to rebase the local branch over the composed commit from #3.
+			//
+			// For the following timings:
+			// +----------------------+-------------+--------------+
+			// |                      | bunchSize:1 | bunchSize:10 |
+			// +----------------------+-------------+--------------+
+			// | localBranchSize: 10  | t1          | t2           |
+			// | localBranchSize: 100 | t3          | t4           |
+			// +----------------------+-------------+--------------+
+			// If op bunching is used, the time taken for each scenario is as follows:
+			// t1 = rebase 1  inbound commit  onto trunk + compose 1  commit  + rebase the local branch of size 10  over one commit
+			// t2 = rebase 10 inbound commits onto trunk + compose 10 commits + rebase the local branch of size 10  over one commit
+			// t3 = rebase 1  inbound commit  onto trunk + compose 1  commit  + rebase the local branch of size 100 over one commit
+			// t4 = rebase 10 inbound commits onto trunk + compose 10 commits + rebase the local branch of size 100 over one commit
+			// Therefore, if op bunching is used, then t4 should be roughly equal to t3 + t2 - t1.
+			for (const bunchSize of bunchSizes) {
+				for (const localBranchSize of localBranchSizes) {
+					const test = benchmark({
+						type: BenchmarkType.Measurement,
+						title: `Rebase ${localBranchSize} local commits over ${bunchSize} inbound commits`,
+						benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>) => {
+							let duration: number;
+							do {
+								// Since this setup collects data from one iteration, assert that this is what is expected.
+								assert.equal(state.iterationsPerBatch, 1);
+
+								const { provider, sender, receiver } = setupSenderAndReceiver();
+								// Send local commits from the receiver but don't sequence them because we want them to be
+								// in the local branch.
+								sendLocalCommits(receiver, localBranchSize, "r");
+								// Send and sequence local commits from the sender. These are the commits that should be
+								// bunched together.
+								sendLocalCommits(sender, bunchSize, "s");
+								sender.containerRuntime.flush();
+
+								const before = state.timer.now();
+								// Resume the receiver and process the bunched commits. This should force the local branch
+								// to be rebased over the bunch.
+								// The receiver will not process its local commits since they were never flushed.
+								receiver.containerRuntime.resumeInboundProcessing();
```

Comment:
My understanding is that calling resumeInboundProcessing here won't lead to any message processing (inbound or outbound). Assuming that's true, I think it would make sense to pull it above the `const before = state.timer.now()` statement.

Reply:
Messages that it received while it was paused will be processed on calling `resumeInboundProcessing`. That's how I have implemented it currently.

Reply:
But it should have received 0 messages so far. It will only receive and process the messages from the sender when `synchronizeMessages` is called. If you put a breakpoint in EditManager's `addSequencedChanges` it will not get hit as a result of calling `resumeInboundProcessing`.

Reply:
oh that's right. I thought you were asking in general and not about this particular scenario.